boards/st/nucleo_h503rb/nucleo_h503rb.dts

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright (c) 2024 Arrow Electronics.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/dts-v1/;
+#include <st/h5/stm32h503Xb.dtsi>
+#include <st/h5/stm32h503rbtx-pinctrl.dtsi>
+#include "arduino_r3_connector.dtsi"
+#include "st_morpho_connector.dtsi"
+#include <zephyr/dt-bindings/input/input-event-codes.h>
+
+
+/ {
+	model = "STMicroelectronics STM32H503RB-NUCLEO board";
+	compatible = "st,stm32h503rb-nucleo";
+
+	#address-cells = <1>;
+	#size-cells = <1>;
+
+	chosen {
+		zephyr,console = &usart3;
+		zephyr,shell-uart = &usart3;
+		zephyr,sram = &sram0;
+		zephyr,flash = &flash0;
+	};
+
+	leds: leds {
+		compatible = "gpio-leds";
+		green_led_2: led_2 {
+			gpios = <&gpioa 5 GPIO_ACTIVE_HIGH>;
+			label = "User LD2";
+		};
+	};
+
+	gpio_keys {
+		compatible = "gpio-keys";
+		user_button: button {
+			label = "User";
+			gpios = <&gpioc 13 GPIO_ACTIVE_HIGH>;
+			zephyr,code = <INPUT_KEY_0>;
+		};
+	};
+
+	aliases {
+		led0 = &green_led_2;
+		sw0 = &user_button;
+		watchdog0 = &iwdg;
+	};
+};
+
+&clk_csi {
+	status = "okay";
+};
+
+&clk_hsi {
+	status = "okay";
+};
+
+&clk_hsi48 {
+	status = "okay";
+};
+
+&clk_hse {
+	clock-frequency = <DT_FREQ_M(24)>;
+	status = "okay";
+};
+
+&pll {
+	div-m = <2>;
+	mul-n = <40>;
+	div-p = <2>;
+	div-q = <2>;
+	div-r = <2>;
+	clocks = <&clk_hse>;
+	status = "okay";
+};
+
+&rcc {
+	clocks = <&pll>;
+	clock-frequency = <DT_FREQ_M(240)>;
+	ahb-prescaler = <1>;
+	apb1-prescaler = <1>;
+	apb2-prescaler = <1>;
+	apb3-prescaler = <1>;
+};
+
+&usart1 {
+	pinctrl-0 = <&usart1_tx_pb14 &usart1_rx_pb15>;
+	pinctrl-names = "default";
+	current-speed = <115200>;
+	status = "okay";
+};
+
+&usart3 {
+	pinctrl-0 = <&usart3_tx_pa4 &usart3_rx_pa3>;
+	pinctrl-names = "default";
+	current-speed = <115200>;
+	status = "okay";
+};
+
+&rng {
+	status = "okay";
```

Comment:
Did you check RNG with related testcase

Reply:
Yes, I ran the following tests for RNG:
`zephyr/scripts/twister --device-testing --device-serial /dev/ttyACM0 --device-serial-baud 115200 -p nucleo_h503rb  -T zephyr/tests/subsys/random/rng`

I got the following output:

```shell
INFO    - JOBS: 16 
INFO    - Adding tasks to the queue... 
INFO    - Added initial list of jobs to queue 
INFO    - Total complete:    5/   5  100%  skipped:    1, failed:    0, error:    0 
INFO    - 5 test scenarios (5 test instances) selected, 1 configurations skipped (0 by static filter, 1 at runtime). 
INFO    - 4 of 5 test configurations passed (100.00%), 0 failed, 0 errored, 1 skipped with 0 warnings in 25.50 seconds 
INFO    - In total 4 test cases were executed, 1 skipped on 1 out of total 1 platforms (100.00%) 
INFO    - 4 test configurations executed on platforms, 0 test configurations were only built. 
```

The test case `drivers.rng.random_psa_crypto` was skipped.